VideoAPI/Testing.Common/Helper/ZAP.cs

```diff
@@ -0,0 +1,168 @@
+                    .AddUserSecrets("9AECE566-336D-4D16-88FA-7A76C27321CD")
+                    .Build()
+                    .GetSection("ZAPConfiguration")
+                    .Get<ZAPConfiguration>();
+            }
+        }
+
+        public static IWebProxy WebProxy
+        {
+            get
+            {
+                return ZapConfiguration.RunZap ? new WebProxy($"http://{ZapConfiguration.ApiAddress}:{ZapConfiguration.ApiPort}", false) : null;
+            } 
+        }
+         
+
+        public static void StartZAPDaemon()
+        {
+            if (!ZapConfiguration.RunZap) return;
+
+            var zapProcessStartInfo = new ProcessStartInfo
+            {
+                FileName = @$"{ZapConfiguration.WorkingDirectory}\ZAP.exe",
+                WorkingDirectory = ZapConfiguration.WorkingDirectory,
+                Arguments = $"-daemon -host {ZapConfiguration.ApiAddress} -port {ZapConfiguration.ApiPort}"
+            };
+             
+            Process.Start(zapProcessStartInfo);
+             
+            CheckIfZAPHasStartedByPollingTheAPI(1);
+        } 
+
+        public static void CheckIfZAPHasStartedByPollingTheAPI(int minutesToWait)
+        { 
+            var watch = new Stopwatch();
+            watch.Start();
+            var millisecondsToWait = ZapConfiguration.MinutesToWait * 60 * 1000;
+            var zapUrlToDownload = $"http://{ZapConfiguration.ApiAddress}:{ZapConfiguration.ApiPort}";
+
+            while (millisecondsToWait > watch.ElapsedMilliseconds)
+            {
+                try
+                {
+                    new WebClient().DownloadString(zapUrlToDownload);
+                    return;
+                }
+                catch
+                {
+                    Thread.Sleep(2000);
+                } 
+            }
+
+            throw new Exception(string.Format("Waited for {0} minute(s), however could not access the API successfully, hence could not verify if ZAP started successfully or not", minutesToWait));
+        }
+
+        public static void Scan(string url)
+        {
+            if (!ZapConfiguration.RunZap) return;
+
+            if (ZapConfiguration.RunSpider)
+            {
+                StartSpidering(url);
+            }
+
+            if (ZapConfiguration.RunActiveScan)
+            {
+                StartActiveScanning(url);
+            }
+        }
+
+        private static void StartSpidering(string target)
+        {
+            _apiResponse = _api.spider.scan(target, "", "", "", "");
+            var scanid = ((ApiResponseElement)_apiResponse).Value;
+            PollTheSpiderTillCompletion(scanid);
+        }
+
+        private static void PollTheSpiderTillCompletion(string scanid)
+        {
+            int spiderProgress;
+            while (true)
+            {
+                Sleep(1000);
+                spiderProgress = int.Parse(((ApiResponseElement)_api.spider.status(scanid)).Value);
```

Comment:
This will throw if its not the expected response or is null. Maybe try TryParse or casring using "As" and checking for null afterwards

Reply:
Updated